pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go

```diff
@@ -55,6 +55,86 @@ type KeycloakAPIRealm struct {
 	// TODO: change to values and use kubebuilder default annotation once supported
 	// +optional
 	AdminEventsDetailsEnabled *bool `json:"adminEventsDetailsEnabled,omitempty"`
+
+	// Authentication flows
+	// +optional
+	AuthenticationFlows *[]KeycloakAPIAuthenticationFlow `json:"authenticationFlows,omitempty"`
+
+	// Authenticator config
+	// +optional
+	AuthenticatorConfig []KeycloakAPIAuthenticatorConfig `json:"authenticatorConfig"`
+}
+
+type KeycloakAPIAuthenticationFlow struct {
+	// Alias
+	Alias string `json:"alias"`
+
+	// Authentication executions TODO: really optional?
+	// +optional
```

Comment:
For every field here, can you comment on which should not be optional? In the javadoc they were marked as optional

Reply:
This causes an NPE if missing. Fixed.

I removed a few of the obvious +optional which are not really optional.

Reply:
Unfortunately, there's no good way to do it. Keycloak [REST API documentation](https://www.keycloak.org/docs-api/10.0/rest-api/index.html) should give you some idea. But it can be 100% trusted.

In the long term, we plan to improve our REST interface. But for the time being we need to stick with what we have.